### PR TITLE
DoubleExt privatekeys: Access all data (Cardano fix)

### LINKED ISSUE #2136: [Cardano] Extended private key contents is not exposed through the C interface
Extended private key has 6*32 bytes; however, only the first 32 bytes can be accessed through the C/TW interface.
In [`PrivateKey` class](https://github.com/trustwallet/wallet-core/blob/master/src/PrivateKey.h) there are different fields for extensions, however these are not accessible through C interface.
Options:
- Expose the concatenated full data, or
- Store in one single field, and create accessors for the different subparts.
 
Discussed in https://github.com/trustwallet/wallet-core/discussions/2135

<div type='discussions-op-text'>

<sup>Originally posted by **kai-newton** April  8, 2022</sup>
    guard let wallet = HDWallet(mnemonic: mnemonic, passphrase: pin) else {
      throw "New base wallet could not be generated"
    }
    
    let walletKey: PrivateKey = wallet.getKeyForCoin(coin: .cardano)

The code above is how I am currently generating private keys for Cardano, however, these keys produce an ``"Invalid extended key"`` error when signing transactions. Any idea of what I'm doing wrong?</div>

### PR DESCRIPTION
**Pull request by catenocrypt**
## Description

Change approach to double-extended private key (used by Cardano):
instead of storing each subpart differently, store them all in the `bytes` member.
Issue was that in some places (through the C interface) only the `bytes` member was accessed, resulting in 32 bytes only (instead of 192 bytes).
Fixes #2136 .

## Testing instructions

Unit tests.

## Types of changes

* Bug fix (non-breaking change which fixes an issue) 

## Checklist

- [x] Prefix PR title with `[WIP]` if necessary.
- [x] Add tests to cover changes as needed.
- [x] Update documentation as needed.
- [x] I have read the [guidelines](https://developer.trustwallet.com/wallet-core/newblockchain#integration-criteria) for adding a new blockchain
- [x] If there is a related Issue, mention it in the description (e.g. Fixes #<issue_number> ).

### CODE REVIEW

**Comment by hewigovens** [comment]
Solana fix?

File: src/PrivateKey.h [hunk 1]
```diff
@@ -20,15 +20,20 @@ class PrivateKey {
...
-    Data secondExtension;
-    Data secondChainCode;
+    Data key() const { return getSubKey(0, 32); }
+    Data extension() const { return getSubKey(32, 32); }
+    Data chainCode() const { return getSubKey(2*32, 32); }
+    Data secondKey() const { return getSubKey(3*32, 32); }
+    Data secondExtension() const { return getSubKey(4*32, 32); }
+    Data secondChainCode() const { return getSubKey(5*32, 32); }
+    // safe subdata utility
+    Data getSubKey(size_t index, size_t size) const;
```
Comment (hewigovens) [inline src/PrivateKey.h:36]: Move to Data.h for future reuse? subData's safe version

**Review by hewigovens — APPROVED** [review]

File: src/PrivateKey.h (hunk 1 above)
Comment (Bastek8) [reply src/PrivateKey.h]: Thanks

**Review by Bastek8 — COMMENTED** [review]
#

**Pull request merged**